### PR TITLE
task was solved

### PR DESCRIPTION
**Pull request by vanyachyzh**

### CODE REVIEW

File: src/formatDate.js [hunk 1]
```diff
@@ -43,14 +43,53 @@
...
- * @param {string[]} fromFormat
+ * @param {string[]} string
  * @param {string[]} toFormat
  *
  * @returns {string}
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const string = fromFormat.join('.');
```
Comment (witflash) [inline src/formatDate.js:53]: Why do you need to join the string? You can check by indexOf inside the original array.

File: src/formatDate.js [hunk 2]
```diff
@@ -43,14 +43,53 @@
...
+  const separator = toFormat[3];
+  const newFormat = [];
+
+  if (fromFormat.includes('YYYY')) {
+    year = date.slice(string.indexOf('YYYY'), string.indexOf('YYYY') + 4);
+  } else {
+    year = date.slice(string.indexOf('YY'), string.indexOf('YY') + 2);
+  }
+
+  for (let i = 0; i < toFormat.length; i++) {
```
Comment (witflash) [inline src/formatDate.js:67]: You don't need to use a cycle for this task.
You can check indexOf year, day and month inside `fromFormat`, get value by this index from `date` and place by index from `toFormat`.

**Review by witflash — CHANGES_REQUESTED** [review]
Check my comments, please

File: src/formatDate.js [hunk 3]
```diff
@@ -43,14 +43,52 @@
...
+
+    return nextDate.join(toFormat[3]);
+  }
+
+  if (!isDouble1) {
+    nextDate[indYear2] = year[2] + year[3];
+  }
+
+  if (!isDouble2) {
+    nextDate[indYear2] = year < 30 ? '20' + year : '19' + year;
```
Comment (VitaliyBondarenko1982) [inline src/formatDate.js:88]: Good job!

Please create constants for all magic numbers and hardcoded strings outside of function in next tasks.

**Review by VitaliyBondarenko1982 — APPROVED** [review]

**Pull request closed without merging**